miso-web/src/main/webapp/scripts/hot_pool.js

```diff
@@ -127,6 +127,25 @@ HotTarget.pool = {
       }));
     }
   },
+  {
+    name: "Avg. Insert Size",
```

Comment:
This doesn't need to be a callback. Just include the calculation result in `PoolDto` and calculate it `Dtos.asDto(Pool)`.

Reply:
Could you expand on this please? I'm still getting to grips with the new Dto pattern.

Reply:
Rather than convert model objects JSON, which tend of have a lot of relationships that aren't relevant, there's a separate set of classes for JSON-serialisation/deserialisation. They mostly match the model classes, but vary in a few important ways:

- they have fewer relationships to other objects
- they have extra fields with the results of calculations
- complex data types are already converted to user-representation (_e.g._, all the dates are converted to strings)

All the data required to calculate the average is already loaded, so rather than make a call that reloads all that data and does a cheap calculation on it, every time a pool is sent to the front end, include the results of that calculation.

The basic process is to add a field to `PoolDto`, probably a `Double` or `String`, with getters and setters. The `asDto` method gets called whenever serialisation is required. It probably makes sense to do `orElse(null)` on the stream to have a number or null and in this case, an average of zero is kind of equivalent to null, rather than storing it as a string.

In the JavaScript, any pool (the actions, the HoT columns, the list columns, will have access to your new field.